client/src/components/Sidebar/menu/SubMenuItem.tsx

```diff
@@ -40,7 +45,21 @@ export const SubMenuItem: React.FunctionComponent<SubMenuItemProps> = ({
   }, [isOpen]);
 
   const onClick = () => {
-    if (!isDismissed(item, state)) {
+    // Only dispatch set-dismissible-state when there are dismissible items
+    // in the submenu and the submenu has not been opened before. Note that
+    // the term "submenu" for this component means that this menu item *has*
```

Comment:
Adding this comment as it can be confusing for those unfamiliar with the code: the term `SubMenuItem` we use actually refers to a `MenuItem` that has sub `MenuItem`s, rather than those sub items.

Reply:
Good call